src/Simplify.cpp

```diff
@@ -247,6 +251,27 @@ void Simplify::ScopedFact::learn_true(const Expr &fact) {
             }
             simplify->bounds_and_alignment_info.push(v->name, expr_info);
             bounds_pop_list.push_back(v);
+        } else if ((div && (v = div->a.as<Variable>())) ||
+                   (div_mul && (v = div_mul->a.as<Variable>()))) {
+            // Learn from expressions of the form (x / a) * b == c
+            // We might know the bounds of the variable.
+            Interval range = solve_for_inner_interval(eq, v->name);
```

Comment:
solve_for_inner_interval is pretty heavy-duty, and might get more expensive in future. Can it be done directly for this form?

Reply:
It can, but it's a fairly tricky bit of logic. Totally doable, I just didn't want to duplicate it. The other thing I thought about doing was only entering this block if a, b and c are constants. In that case, solve_for_inner_interval should be pretty cheap right?

Reply:
I think this is true (z3 can't find counterexamples at least):

```(x / c0) * c1 == c2```
is equivalent to:
``` x >= c0 * (c2/c1) && x < c0 * (c2 / c1 + 1) ```
when
```c0 > 0 && c1 > 0 && c2 % c1 == 0```

Reply:
I found more small simple equations that needed support. Let me know what you think of the current thing. If we don't want to use solve here, I suspect we either need to add a "solve_constant" helper, or prevent situations like this in the first place by fixing specialize, but I've spent quite a bit of effort on that approach without success.